course_discovery/apps/publisher/signals.py

```
@@ -44,40 +44,20 @@ def create_course_run_in_studio_receiver(sender, instance, created, **kwargs):
 
         logger.info('Publishing course run [%d] to Studio...', instance.id)
         api = StudioAPI(instance.course.partner.studio_api_client)
-
         discovery_course_run = instance.discovery_counterpart_latest_by_start_date
 
-        if discovery_course_run:
-            try:
-                logger.info('Creating a re-run of [%s]...', discovery_course_run.key)
-                response = api.create_course_rerun_in_studio(instance, discovery_course_run)
-            except SlumberBaseException as ex:
-                logger.exception(
-                    'Failed to create course re-run [%s] on Studio: %s',
-                    discovery_course_run.key,
-                    ex.content
-                )
-                raise
-        else:
-            try:
-                logger.info('Creating a new run of [%s]...', instance.course.key)
-                response = api.create_course_run_in_studio(instance)
-            except SlumberBaseException as ex:
-                logger.exception('Failed to create course run [%s] on Studio: %s', instance.id, ex.content)
-                raise
+        try:
+            response = api.push_to_studio(instance, create=True, old_course_run=discovery_course_run)
+        except SlumberBaseException as ex:
+            logger.exception('Failed to create course run [%s] on Studio: %s', course.key, ex.content)
+            raise
+        except Exception:  # pylint: disable=broad-except
```

I think we should be cautious with broad exception catches, are there specific exceptions we are looking for here?

---

This specific instance is actually a tightening of a bare-except in the old code that I was refactoring. I'm not sure what all the exceptions it expects or cares about are. Probably just SlumberBaseException in practice.

I *suspect* the original bare except was added simply so they can log a message that tests can pick up. They have one test that adds a side effect exception and checks that it was caught. :man_shrugging: I'd be fine with dropping this except (which simply logs and raises anyway) and reworking that test.